packages/analytics-core/src/plugins/destination.ts

```diff
@@ -284,34 +302,50 @@ export class Destination implements DestinationPlugin {
       this.config.loggerProvider.warn(getResponseBodyString(res));
     }
 
-    this.addToQueue(...retry);
+    const tryable = this.tryableList(retry);
+    this.scheduleTryable(tryable);
   }
 
   handleOtherResponse(list: Context[]) {
-    this.addToQueue(
-      ...list.map((context) => {
-        context.timeout = context.attempts * this.retryTimeout;
-        return context;
-      }),
-    );
+    const later = list.map((context) => {
+      context.timeout = context.attempts * this.retryTimeout;
+      return context;
+    });
+
+    const tryable = this.tryableList(later);
+    this.scheduleTryable(tryable);
+    this.saveEvents();
```

Comment:
Seems no need to call `saveEvents()` here like other response handlers.

Reply:
Deleted in https://github.com/amplitude/Amplitude-TypeScript/pull/689/commits/3b954b4c61f9715be3dfff3ed6fb7e9bdc7d4d35